staging/src/k8s.io/apimachinery/pkg/util/httpstream/wsstream/conn.go

```diff
@@ -256,6 +268,21 @@ func (conn *Conn) handle(ws *websocket.Conn) {
 		if len(data) == 0 {
 			continue
 		}
+		if supportsStreamClose && data[0] == remotecommand.StreamClose {
+			if len(data) != 2 {
+				klog.Errorf("Single channel byte should follow stream close signal. Got %d bytes", len(data)-1)
+				break
+			} else {
+				channel := data[1]
+				if int(channel) >= len(conn.channels) {
+					klog.Errorf("Channel out of range", channel)
+					break
+				}
+				klog.V(4).Infof("Received half-close signal from client; close %d stream", channel)
+				conn.channels[channel].Close() // After first Close, other closes are noop.
```

Comment:
after we get this Close(), is it an error to get any other messages trying to send us data on this stream (e.g. reaching line 291 below with this channel)? do we need to detect that and avoid calling DataFromSocket on a channel that a close event was sent to?

Reply:
After the `Close()`, we will get an `EOF` or error on the next message receive in the loop. It would not be possible to get to `DataFromSocket` after the `Close()`.